src/gamebryo/gamebryosavegame.cpp

```diff
@@ -123,19 +123,52 @@ void GamebryoSaveGame::FileWrapper::setPluginString(StringType type)
   m_PluginString = type;
 }
 
-void readQDataStream(QDataStream& data, void* buff, std::size_t length)
+void GamebryoSaveGame::FileWrapper::readQDataStream(QDataStream& data, void* buff,
+                                                    std::size_t length)
 {
-  int read = data.readRawData(static_cast<char*>(buff), static_cast<int>(length));
-  if (read != length) {
+  int read    = data.readRawData(static_cast<char*>(buff), static_cast<int>(length));
+  bool result = true;
+  if (read != length && m_CompressionType == 1) {
+    bool result = readNextChunk();
+    if (result) {
+      read += data.readRawData(static_cast<char*>(buff) + read,
+                               static_cast<int>(length - read));
+    }
+  }
+  if (read != length || !result) {
     throw std::runtime_error("unexpected end of file");
   }
 }
 
 template <typename T>
-void readQDataStream(QDataStream& data, T& value)
+void GamebryoSaveGame::FileWrapper::readQDataStream(QDataStream& data, T& value)
 {
-  int read = data.readRawData(reinterpret_cast<char*>(&value), sizeof(T));
-  if (read != sizeof(T)) {
+  int read    = data.readRawData(reinterpret_cast<char*>(&value), sizeof(T));
```

Comment:
I think this should re-use the `readQDataStream` above, as far as I can tell, this is equivalent to `readQDataStream(data, reinterpret_cast<char*>(&value), sizeof(T))`.

Reply:
I would also add a `static_assert` or SFINAE parameter to check that `T` is trivial and standard layout, otherwise this kind of operation can wreck havoc.

This should work:

```
#include <type_traits> // if needed

static_assert(std::is_trivial_v<T> && std::is_standard_layout_v<T>);
```

Reply:
Oh, I see what you mean. I think that would work, yeah.